src/steering_model.py

```diff
@@ -28,3 +31,46 @@ def __init__(self, path_to_onnx_model):
 
     def predict(self, input_frame):
         return self.session.run(None, { self.input_name: input_frame })[0]
+
+
+class SteeringModel:
+
+    def __init__(self, path_to_onnx_model):
+        self.steering_model = OnnxModel(path_to_onnx_model)
+
+    def predict(self, input_frame, car):
+        return self.steering_model.predict(input_frame).item(), None
+
+
+class ConditionalSteeringModel(SteeringModel):
+
+    def __init__(self, path_to_steering_model, path_to_speed_model):
+        super().__init__(path_to_steering_model)
+        self.speed_model = OnnxModel(path_to_speed_model) if path_to_speed_model else None
+
+    def predict(self, input_frame, car):
+        predictions = self.steering_model.predict(input_frame)
+        steering_angle = predictions[0][car.human_turn_signal].item()
+        speed = self.speed_model.predict(input_frame)[0].item() if self.speed_model else None
+        return steering_angle, speed
+
+
+class ConditionalWaypointsModel(SteeringModel):
+
+    def __init__(self, path_to_steering_model, path_to_speed_model, num_waypoints=4, ref_distance=6.8,
+                 use_vehicle_pos=True, latitudinal_correction=-0.30):
```

Comment:
Is the latitudinal correction related to camera's position (off from the center by 30cm)? 

Can we use the info from the trace's `params.xml` file for this instead of passing it as an argument? Ideally, we'd then have a single source of truth for all camera-position stuff.

Reply:
I don't know what exactly is causing this systematic bias. It could be the camera position, could be something else. In the long run better controller should be used anyway, so I don't think it matters too much in this case.

Reply:
As I understand this is unexplained bias of the model and fixed by the correction. It's ok, as long as it affects only WaypointsModel and is commented. I would rename the parameter `lateral_correction`, which is what you probably meant.

Reply:
Renamed.